common/src/main/java/com/microsoft/identity/common/internal/migration/TokenCacheItemMigrationAdapter.java

```diff
@@ -185,7 +185,18 @@ public static boolean tryFociTokenWithGivenClientId(@SuppressWarnings(WarningTyp
         final MicrosoftStsOAuth2Strategy strategy = new MicrosoftStsOAuth2Strategy(config, strategyParameters);
 
         final String refreshToken = refreshTokenRecord.getSecret();
-        final String scopes = BaseController.getDelimitedDefaultScopeString();
+
+        final String scopes;
+        // Hardcoding Teams Agent's client ID with the scope it's pre-authorized for.
+        // This is because if only the default scope is passed, eSTS will set the resource ID (on its side)
+        // based on the RT (Which the given clientId might not be pre-authorized for).
+        // TODO: make pre-authorization of MSGraph User.read (and the default scopes) a requirement
+        //       for every FoCI apps (and hardcode it here).
+        if (TextUtils.equals(clientId, "87749df4-7ccf-48f8-aa87-704bad0e0e16")) {
```

Comment:
Can we add a constant for this?

Reply:
This is temporary and should be removed in the next release. (once we verify that using MSGraph user.read doesn't break anything) - because of that I prefer putting all of them in one place.